src/transformStateWithClones.js

```diff
@@ -7,7 +7,36 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const result = [];
+  let currentState = { ...state };
+
+  for (const act of actions) {
+    let modifiedState = { ...currentState };
+
+    switch (act.type) {
+      case 'addProperties':
+        Object.assign(modifiedState, act.extraData);
+        break;
+
+      case 'removeProperties':
+        for (const key of act.keysToRemove) {
+          delete modifiedState[key];
+        }
+        break;
+
+      case 'clear':
+        modifiedState = {};
+        break;
+
+      default:
+        return 'error';
```

Comment:
```suggestion
        throw new Error(`Invalid action type: ${action.type}`);
```
It doesn't make much sense to return an error. You should throw it and the message should be informative